content/code/testing.md

```diff
@@ -104,6 +150,20 @@ This can be very useful, as it enables creating a hierarchical structure within
 {!index_internal_test.md!}
     ```
 
+##### Structure
+
+In those tests, usually the structure is composed of:
+
+* At the top-level, the common testing variables that are used in multiple subtests.
+    * This can be a logger that discards logs, some test input that can be reused, expected return values shared between tests, and so on.
+* In each sub-test:
+    * The test is defined as parallelized;
+    * The inputs and outputs are defined in variables;
+    * The mocks are defined and their methods overridden when necessary;
+    * If test is for a method, the method's struct (the subject of the test) is created and given the mocks;
+    * The tested method is called on the subject, its return values are often stored in `got` and `err`: `got, err := mything.Do(input)`;
+    * Assertions are made on the returned values.
```

Comment:
It's also nice to use `require.NoError(t, err)` on the `nominal case` tests. This will insure that test will fail before the assertions.

Reply:
It's what I implied by using `values` plural, but you're right maybe it's not clear enough. WDYT of

```suggestion
	* If the test should not error, `require.NoError` is called on the error return;
    * Assertions are made on the returned value.
```